### PR TITLE
Mcs 285: Optimal path planner should account for agent reach

### PR DESCRIPTION
**Pull request by chrislong**

### CODE REVIEW

File: scene_generator/interaction_goals.py [hunk 1]
```diff
@@ -150,12 +151,43 @@ def get_navigation_actions(start_location: Dict[str, Any],
...
+    step_dist = 0
+    dist_moved = 0
+    while total_distance < MAX_REACH_DISTANCE and i > 0:
+        i -= 1
+        action = actions[i]
+        if action['action'] != 'MoveAhead':
+            break
+        step_dist = action['params']['amount'] * MAX_MOVE_DISTANCE
+        total_distance += step_dist
+        dist_moved += step_dist
```
Comment (ThomasSchellenbergNextCentury) [inline scene_generator/interaction_goals.py:184]: In what scenarios will we need to remove more than one Move action from the end of the action list?

File: scene_generator/interaction_goals.py [hunk 2]
```diff
@@ -150,12 +151,43 @@ def get_navigation_actions(start_location: Dict[str, Any],
...
+
+def trim_actions_to_reach(actions: List[Dict[str, Any]],
+                          performer: Tuple[float, float],
+                          heading: float,
+                          goal_obj: Dict[str, Any]) -> \
+                          Tuple[List[Dict[str, Any]], Tuple[float, float]]:
+    """Trim the action list from the end so that the performer doesn't
+    take additonal steps toward the goal object once they're within
+    MAX_REACH_DISTANCE.
+    """
```
Comment (ThomasSchellenbergNextCentury) [inline scene_generator/interaction_goals.py:170]: This doesn't appear to reduce the `amount` of the last Move action, like I'd have expected.  Is that not necessary?

File: scene_generator/interaction_goals.py (hunk 1 above)
Comment (chrislong) [reply scene_generator/interaction_goals.py:184]: Reach is greater than movement, and the last Move action might be for a small amount, so removing it and the previous action may sometimes still leave the agent within reach.

File: scene_generator/interaction_goals.py (hunk 2 above)
Comment (chrislong) [reply scene_generator/interaction_goals.py:170]: I thought it was sufficient to minimize the number of actions, not necessarily the distance traversed, but I could change it.

Comment (deanwetherby) [reply scene_generator/interaction_goals.py:170]: Number of actions is what we are trying to reduce so reducing the amount isn't necessary although there's a fun optimization to consider the following action in order to be really optimal!

Comment (chrislong) [reply scene_generator/interaction_goals.py:170]: I think I see what you mean. At least for some value of "fun".

File: scene_generator/interaction_goals.py (hunk 1 above)
Comment (ThomasSchellenbergNextCentury) [reply scene_generator/interaction_goals.py:184]: Oh, reach is greater than movement.  I had forgotten that.  Thanks for the explanation.

File: scene_generator/interaction_goals.py (hunk 2 above)
Comment (ThomasSchellenbergNextCentury) [reply scene_generator/interaction_goals.py:170]: That makes sense.  Thanks.

**Pull request merged**